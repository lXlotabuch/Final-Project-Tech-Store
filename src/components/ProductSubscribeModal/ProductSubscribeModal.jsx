import React from 'react';
import { connect } from 'react-redux';
import {
  Modal,
  Form,
  Input,
  message
} from 'antd';
import { hideSubscribeModal } from '../../store/subscriceOnProductModal/middleware'
import { selectIsOpenModal } from '../../store/subscriceOnProductModal/reducer'
import { selectCustomerInfo } from '../../store/customer/reducer'

const mapStateToProps = (state) => ({
  isOpen: selectIsOpenModal(state),
  customerInfo: selectCustomerInfo(state)
})

const ProductSubscribeModal = connect(mapStateToProps, { hideSubscribeModal })((
  {
    isOpen,
    hideSubscribeModal,
    customerInfo,
  }
) => {
  const [form] = Form.useForm()

  const onSubmit = () => {
    form.resetFields()
    hideSubscribeModal()
    message.success('Thank you, you will be notified when this item is in stock!')
  };
  const onCancel = () => {
    form.resetFields()
    hideSubscribeModal()
  }
  const fields = [
    {
      name: 'email',
      value: customerInfo.email
    }
  ]

  return (
    <Modal
      title="Notify on product availability"
      visible={isOpen}
      okText="Subscribe"
      onCancel={onCancel}
      okButtonProps={{htmlType: 'submit', form: 'subscribeOnProductsForm', shape: 'round'}}
      cancelButtonProps={{shape: 'round'}}
      destroyOnClose
    >
      <Form name="subscribeOnProductsForm" onFinish={onSubmit} form={form} fields={fields}>
        <Form.Item
          label="Email"
          name="email"
          rules={[
            {
              required: true,
              message: 'Please input your email.',
            },
            {
              type: 'email',
              message: 'Entered data is not an email.',
            },
          ]}
        >
          <Input placeholder="Email" />
        </Form.Item>
      </Form>
    </Modal>
  );
})
export default ProductSubscribeModal