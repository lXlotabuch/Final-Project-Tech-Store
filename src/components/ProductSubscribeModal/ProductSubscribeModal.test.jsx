/* eslint-disable no-unused-vars */
import { shallow } from 'enzyme'
import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { store } from '../../store/index'
import ProductSubscribeModal from './ProductSubscribeModal'

describe('All tests for productSubscribeModal', () => {
  test('Should have simple render', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ProductSubscribeModal />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('Should have the form', () => {
    const result = shallow(
      <Provider store={store}>
        <ProductSubscribeModal />
      </Provider>
    )
    expect(result.find('#subscribeOnProductsForm')).toBeDefined()
  })
})