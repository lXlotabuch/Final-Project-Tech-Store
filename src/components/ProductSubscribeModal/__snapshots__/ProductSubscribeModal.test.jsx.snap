// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All tests for productSubscribeModal Should have simple render 1`] = `<DocumentFragment />`;
